### PR TITLE
Error handler

### PR DESCRIPTION
**Pull request by ovekyc**

### CODE REVIEW

**Review by hewonjeong — APPROVED** [review]

File: src/server/images/controllers.js [hunk 1]
```diff
@@ -26,10 +26,7 @@ export default {
     })
     .catch((err)=>{
       if (err.notFound) {
-        cb(new APIError(err, {
-          statusCode: 400,
-          message: 'Item was not found'
-        }));
+        cb(new APIError(err, 400)); // 'Item was not found'
```
Comment (festinalente91) [inline src/server/images/controllers.js]: I tested leveldb not found error and there is no err.message property.

Comment (festinalente91) [reply src/server/images/controllers.js]: There is just err.notFound property.

**Comment by festinalente91** [comment]
I think NotFoundError class is a great idea.
However it seems unnecessary, because when you extends some class there have to be a specific reason, but in this case you just used the message property of super class which is Error class. So in my opinion using constant is better in this case. :)

**Comment by ovekyc** [comment]
I think It can be expanded. like having errno or something another.
(errno for recognize which error, not using like message)
So I think Error class is better than just string.

**Comment by festinalente91** [comment]
I agree with the point that Error class is better than string because it can have something else :) However, what I want to say is that, the class you defined is just using the message. So why don't you just do it like 
```
const ERROR_MESSAGE = { NOT_FOUND: 'Error : NotFound'} 
return next(new APIError(new Error(ERROR_MESSAGE.NOT_FOUND), 400));
// or return next (new APIError(new Error(ERROR_MESSAGE.NOT_FOUND), 400));
```

**Comment by ovekyc** [comment]
I understand. But i think class is better.
There is already ERRNO constant in ErrorHandler.js file. And it used like
`err = new Error(); err.errno = ERRNO.SOMETHING;`
If there is a lot of various Errors and every error managed with several constant such as MESSAGE, NUMBER, then it maybe hard to read or maintain.

However, if we packaging message, number,... in one class, there is no problem like above.

**Comment by festinalente91** [comment]
I think using a constant is much declarative and simpler than a class.
If you manage the error related constants in the related files, it can improve maintainability.
I think the main reason we are trying to extends Error class is to deal with REST layer's error which includes status code. APIError class already can cover this.

**Comment by ovekyc** [comment]
I did a bit research about it. And i found a blog.
The writer explained about Exception handling and he used enum for error. (which is basically similar with constant of JS)
But on the comment, He said like class is better if we want to take this far. (If i correctly understand lol)

By Owen
`This is a good approach although I think one exception class might be a bit far, you could consider creating a base class and then sub exception classes with different Enums. This would allow you to have componentised development but still use the same base mechanism for handling errors.`
`Also a very minor point and I’m sure just used for an example but CREDIT_CARD_EXPIRED is not really an exception as this is an expected and predictable state for a credit card to be in and should be handled by the “Main Path” of your program.
This is my opinion, of course, but Exceptions should only be used for Exceptional circumstances (see a dictionary definition) otherwise you can end up throwing exceptions up several layers of your application because it is easy or using exception for logical flows.`

By Dele Taylor
`Owen — you’re right that a single exception class is taking things far. Probably too far for most Java developers to stomach. For the skeptical, I say try to keep an open mind. I’ve had a surprising amount of success with the single exception approach over the years.`
`Having said that, I do like your recommendation. It’s a good middle ground. Most developers can accept it and it probably won’t break any of my production code 🙂 As long as you’re recommending a SystemException subclass per subsystem (accounts, inventory, tickets) and not the class-per-error we see today, then I’m on board.`

https://northconcepts.com/blog/2013/01/18/6-tips-to-improve-your-exception-handling/

**Pull request merged**